### PR TITLE
Update Developer Guide #411

### LINKED ISSUE #411: Developer Guide Typo
Under Developer Guide- Setting up - Importing the project into IntelliJ - point12 - point III

> Expand Defaults, and ensure that JUnit is selected

However, Windows 2018.2.2 version shows "Templates" instead of "Defaults"
capture

### PR DESCRIPTION
**Pull request by YezhongZ**
Fixes #411 

DeveloperGuide: update instructions on modifying default test configuration

As of IntelliJ IDEA Community 2018.2, `Defaults` has been renamed to
`Templates`.

### CODE REVIEW

**Comment by pyokagan** [comment]
@yamgent Was this a typo or did IntelliJ just change the name in later versions?

**Comment by pyokagan** [comment]
OK, answering my own question: the outdated screenshot in https://www.jetbrains.com/help/idea/changing-default-run-debug-configurations.html strongly suggests that the name was changed in later versions.

**Comment by pyokagan** [comment]
Seems that this change is fairly recent as well. IntelliJ community 2018.1 still calls it "Default", so this naming change was very likely done in IntelliJ commuity 2018.2.

**Comment by damithc** [comment]
Should we go with `Templates (or Defaults)` to cover both cases?

**Comment by pyokagan** [comment]
@YezhongZ Thanks for the PR. You'll need to add some information to your commit message to adhere to our [commit message conventions](https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message).

In particular, the commit message needs to tell us why you think the change is a good idea.

Normally, I would recommend authors just copy the [issue description](https://github.com/se-edu/addressbook-level2/issues/411#issue-357576549), but in this case it is not really a typo -- the developer guide just needs to be updated with the changes in the latest version of IntelliJ. So, I'd recommend a commit message like this:

```
DeveloperGuide: update instructions on modifying default test configuration

As of IntelliJ IDEA Community 2018.2, `Defaults` has been renamed to
`Templates`.
```

**Comment by pyokagan** [comment]
> Should we go with Templates (or Defaults) to cover both cases?

Yeah, that can be done. In this case, we also mention the old name `Defaults` so that readers who are using an older version of IntelliJ (before 2018.2) will not be confused.

@YezhongZ If you take @damithc 's suggestion, make sure you update the commit message as well.

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -33,7 +33,7 @@
 . Observe how some tests fail. That is because they try to access the test data from the wrong directory (the working directory is expected to be the root directory, but IntelliJ runs the test with `test\` as the working directory by default). To fix this issue:
 .. Go to `Run` -> `Edit Configurations...`
 .. On the list at the left, expand `JUnit`, and remove all the test configurations (e.g. `All in test`) by selecting it and clicking on the '-' icon at the top of the list
-.. Expand `Defaults`, and ensure that `JUnit` is selected
+.. Expand `Templates` (or `Defaults` for version before 2018.2), and ensure that `JUnit` is selected
```
Comment (pyokagan) [inline docs/DeveloperGuide.adoc]: I think readers may wonder which "version" we are talking about here.

Maybe we can make this clearer (though lengthier) by writing:

```
or `Defaults` if you are using an older version of IntelliJ before 2018.2
```
? What do you think?

Comment (YezhongZ) [reply docs/DeveloperGuide.adoc]: Sure. I think your version is clearer to the user!

**Comment by pyokagan** [comment]
@YezhongZ 

Thanks, the changes look good.

Your commit message is currently like this:

```
Update DeveloperGuide:
Update instructions on modifying default test configuration
As of IntelliJ IDEA Community 2018.2,
`Defaults` has been renamed to `Templates
```

Since there is no blank line between the first line (`Update DeveloperGuide:`) and the rest of the message, Git thinks that the commit subject is the entire message.

You'll need to change your commit message to look something like this:

```
DeveloperGuide: update instructions on modifying default test configuration

As of IntelliJ IDEA Community 2018.2, `Defaults` has been renamed to
`Templates`.
```

Note the blank line between the commit subject (`DeveloperGuide: update instructions...`) and the rest of the commit message (`As of IntelliJ IDEA Community...`)

Also note that the commit subject has to be on one line no matter how long it is.

Could you fix this?

**Comment by YezhongZ** [comment]
> @YezhongZ
> 
> Thanks, the changes look good.
> 
> Your commit message is currently like this:
> 
> ```
> Update DeveloperGuide:
> Update instructions on modifying default test configuration
> As of IntelliJ IDEA Community 2018.2,
> `Defaults` has been renamed to `Templates
> ```
> Since there is no blank line between the first line (`Update DeveloperGuide:`) and the rest of the message, Git thinks that the commit subject is the entire message.
> 
> You'll need to change your commit message to look something like this:
> 
> ```
> DeveloperGuide: update instructions on modifying default test configuration
> 
> As of IntelliJ IDEA Community 2018.2, `Defaults` has been renamed to
> `Templates`.
> ```
> Note the blank line between the commit subject (`DeveloperGuide: update instructions...`) and the rest of the commit message (`As of IntelliJ IDEA Community...`)
> 
> Also note that the commit subject has to be on one line no matter how long it is.
> 
> Could you fix this?

Fixed.
Thank you for much for the guidance. I learnt a lot from this PR process.

**Comment by pyokagan** [comment]
@YezhongZ Thanks, I think v5, as it was submitted, is good enough. I'll merge the archive.

In the future, avoid modifying the branch after submitting it for review.

**Comment by pyokagan** [comment]
Merged as f34759a390e1fd8f88144eded5a7897fe7289c8d

**Comment by HEARTOFAL1ON** [comment]
Interesting, I learnt something from this too~ :)

**Pull request closed without merging**